client/consul.go

```diff
@@ -161,7 +163,7 @@ func (c *ConsulService) Deregister(task *structs.Task, allocID string) error {
 		}
 		c.logger.Printf("[INFO] consul: deregistering service %v with consul", service.Name)
```

Comment:
You should move this after the if so that there is only one debug message. Either a successful deregister or an error deregistering same for registering